pkg/tbtc/tbtc.go

```diff
@@ -146,16 +146,69 @@ func Initialize(
 				return
 			}
 
+			confirmationBlock := event.BlockNumber + dkgStartedConfirmationBlocks
+
 			logger.Infof(
-				"DKG started with seed [0x%x] at block [%v]",
+				"observed DKG started event with seed [0x%x] and "+
+					"starting block [%v]; waiting for block [%v] to confirm",
 				event.Seed,
 				event.BlockNumber,
+				confirmationBlock,
 			)
 
-			node.joinDKGIfEligible(
-				event.Seed,
-				event.BlockNumber,
-			)
+			err := node.waitForBlockHeight(ctx, confirmationBlock)
+			if err != nil {
+				logger.Errorf("failed to confirm DKG started event: [%v]", err)
+				return
+			}
+
+			dkgState, err := chain.GetDKGState()
+			if err != nil {
+				logger.Errorf("failed to check DKG state: [%v]", err)
+				return
+			}
+
+			if dkgState == AwaitingResult {
+				// Fetch all past DKG started events starting from one
+				// confirmation period before the original event's block.
```

Comment:
And we do it because if there was a reorg, the event we received could be moved to a block with a lower number than the one we received, correct? Can we clarify it in this comment?

Reply:
See e89b0059